### PR TITLE
CEDS-1914 - NACT code extraction

### PR DESCRIPTION
**Pull request by SteveSugden**
This is very similar to the recent extraction of the TARIC codes page.

The main difference is that removing NACT codes from the ItemType page now leaves that page with only one text field and I was able to simplify the ItemTypeController to remove all of the add/remove item functionality.

The next and final ticket in this series will be to rename ItemType to StatisticalValue and to skip that page when on a simplified dec journey.

### CODE REVIEW

File: app/forms/declaration/ItemType.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import play.api.libs.json.Json
+import utils.validators.forms.FieldValidator._
+
+case class ItemType(statisticalValue: String)
+
+object ItemType {
+
+  implicit val format = Json.format[ItemType]
+
+  val statisticalValueKey = "statisticalValue"
```
Comment (PatrykRudnicki) [inline app/forms/declaration/ItemType.scala]: You don't need this key

File: app/forms/declaration/NactCode.scala [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import services.NationalAdditionalCode
+import utils.validators.forms.FieldValidator._
+
+case class NactCode(nactCode: String)
+
+object NactCode extends DeclarationPage {
+
+  implicit val format = Json.format[NactCode]
+
+  val nactCodeKey = "nactCode"
```
Comment (PatrykRudnicki) [inline app/forms/declaration/NactCode.scala]: You don't need this key

File: conf/messages.en [hunk 3]
```diff
@@ -640,14 +633,25 @@ declaration.cusCode.error.empty = Enter a CUS code
...
+declaration.taricAdditionalCodes.header = Is there an additional TARIC code?
 declaration.taricAdditionalCodes.header.hint = This will be the four digits that come after the commodity code in the {0}. If no additional codes are required, leave blank.
+declaration.taricAdditionalCodes.header.tradeTariff.link = UK Trade Tariff
 declaration.taricAdditionalCodes.error.empty = Enter a TARIC code
 declaration.taricAdditionalCodes.error.length = TARIC code must be exactly 4 characters long
 declaration.taricAdditionalCodes.error.specialCharacters = TARIC code cannot contain specialCharacters
 declaration.taricAdditionalCodes.add.hint = Add TARIC code
 declaration.taricAdditionalCodes.remove.hint = Remove TARIC code {0}
 
+declaration.nationalAdditionalCode.header = Do these goods require a national additional (NACT) code?
```
Comment (PatrykRudnicki) [inline conf/messages.en]: But in the UI we're using `National additional code` so we can use this as variable. but it's okay

File: conf/messages.en [hunk 4]
```diff
@@ -640,14 +633,25 @@ declaration.cusCode.error.empty = Enter a CUS code
...
+declaration.nationalAdditionalCode.header.hint = You can find this in ‘Export measures and restrictions’ tab in the {0}.  If no additional code is required, leave blank.
```
Comment (PatrykRudnicki) [inline conf/messages.en]: Double space in this message

File: test/unit/controllers/declaration/ItemTypeControllerSpec.scala [hunk 5]
```diff
@@ -97,112 +95,42 @@ class ItemTypeControllerSpec extends ControllerSpec with ErrorHandlerMocks with
...
-        verify(mockItemTypePage, times(0)).apply(any(), any(), any(), any())(any(), any())
-      }
-
-      "correct item type is added during saving" in {
+      "valid data is posted" in {
 
         withNewCaching(aDeclaration(withItem(anItem(withItemId(itemId)))))
 
-        val correctForm = Seq(("nationalAdditionalCode", "VATR"), ("statisticalValue", "1234"), saveAndContinueActionUrlEncoded)
+        val badData = Json.toJson(ItemType("7"))
```
Comment (PatrykRudnicki) [inline test/unit/controllers/declaration/ItemTypeControllerSpec.scala]: `wrongData` sounds better?

File: conf/messages.en (hunk 4 above)
Comment (PatrykRudnicki) [reply conf/messages.en]: After {0}. and If

Comment (SteveSugden) [reply conf/messages.en]: You not part of the double-space-after-full-stop debate?   I think the browser ignores it anyway but I'll remove.

File: app/forms/declaration/NactCode.scala (hunk 2 above)
Comment (SteveSugden) [reply app/forms/declaration/NactCode.scala]: These keys are used as constants for field name in the view and tests

File: conf/messages.en (hunk 4 above)
Comment (PatrykRudnicki) [reply conf/messages.en]: No, I hate double spaces everywhere :)

Comment (SteveSugden) [reply conf/messages.en]: That's because you are young and have never used a real typewriter (I suspect)

**Comment by bambuchaAdm** [comment]
Please add temper monkey scripts

**Pull request merged**